sources/packages/backend/apps/queue-consumers/src/processors/virus-scan/virus-scan.processor.ts

```diff
@@ -0,0 +1,67 @@
+import { Process, Processor } from "@nestjs/bull";
+import { CustomNamedError, QueueNames } from "@sims/utilities";
+import { StudentFileService } from "../../services";
+import { Job } from "bull";
+import {
+  VirusScanQueueInDTO,
+  VirusScanResult,
+} from "@sims/services/queue/dto/virus-scan.dto";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import { NotFoundException } from "@nestjs/common";
+import { logProcessSummaryToJobLogger } from "../../utilities";
+import { UNABLE_TO_SCAN_FILE } from "../../constants/error-code.constants";
+
+@Processor(QueueNames.VirusScanProcessor)
+export class VirusScanProcessor {
+  constructor(private readonly studentFileService: StudentFileService) {}
+
+  /**
+   * Perform virus scanning for all the files having pending scan status.
+   * @param job information to perform the process.
+   * @returns VirusScanResult virus scan result log.
+   */
+  @Process()
+  async performVirusScan(
+    job: Job<VirusScanQueueInDTO>,
+  ): Promise<VirusScanResult> {
+    const processSummary = new ProcessSummary();
+    let isInfected: boolean;
+    processSummary.info("Starting virus scan.");
```

Comment:
set a default false value, as even the scanfile fails, it will return false than undefined.

Reply:
the call to `scanFile` in the line 35 will always result in returning `isInfected` as one of the following: `true / false / undefined / null` all of which are handled.